File: RepoPromptTests/CodeMap/Fixtures/cs/smoke.cs
// Smoke test fixture for C# codemap extraction
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace UserManagement
{
    /// <summary>
    /// User role enumeration
    /// </summary>
    public enum UserRole
    {
        Admin,
        Editor,
        Viewer,
        Guest
    }

    /// <summary>
    /// User data model
    /// </summary>
    public class User
    {
        public string Id { get; set; }
        public string Name { get; set; }
        public string Email { get; set; }
        public UserRole Role { get; set; } = UserRole.Guest;
        public DateTime CreatedAt { get; init; }

        public User(string id, string name, string email)
        {
            Id = id;
            Name = name;
            Email = email;
            CreatedAt = DateTime.UtcNow;
        }

        public string DisplayName => $"{Name} ({Role})";

        public bool Validate()
        {
            return !string.IsNullOrEmpty(Id) && !string.IsNullOrEmpty(Name);
        }
    }

    /// <summary>
    /// Data provider interface
    /// </summary>
    public interface IDataProvider<T>
    {
        Task<IEnumerable<T>> FetchAsync();
        Task SaveAsync(T item);
        Task<bool> DeleteAsync(string id);
    }

    /// <summary>
    /// Service configuration
    /// </summary>
    public record Config(int MaxUsers = 1000, int TimeoutSeconds = 30);

    /// <summary>
    /// User service for managing users
    /// </summary>
    public class UserService : IDataProvider<User>, IDisposable
    {
        private readonly Dictionary<string, User> _users = new();
        private readonly Config _config;
        private bool _disposed;

        public static int MaxUsers => 1000;
        public static readonly TimeSpan DefaultTimeout = TimeSpan.FromSeconds(30);

        public UserService(Config? config = null)
        {
            _config = config ?? new Config();
        }

        public int Count => _users.Count;

        public async Task<IEnumerable<User>> FetchAsync()
        {
            await Task.Yield();
            return _users.Values.ToList();
        }

        public async Task SaveAsync(User user)
        {
            if (user == null)
                throw new ArgumentNullException(nameof(user));
            
            if (!user.Validate())
                throw new ArgumentException("Invalid user data", nameof(user));
            
            if (_users.Count >= _config.MaxUsers)
                throw new InvalidOperationException("Max users reached");

            await Task.Yield();
            _users[user.Id] = user;
        }

        public async Task<bool> DeleteAsync(string id)
        {
            await Task.Yield();
            return _users.Remove(id);
        }

        public User? Find(string id)
        {
            return _users.GetValueOrDefault(id);
        }

        public IEnumerable<User> GetByRole(UserRole role)
        {
            return _users.Values.Where(u => u.Role == role);
        }

        public void Dispose()
        {
            if (_disposed) return;
            _users.Clear();
            _disposed = true;
            GC.SuppressFinalize(this);
        }
    }

    /// <summary>
    /// User factory with static methods
    /// </summary>
    public static class UserFactory
    {
        private static int _counter;

        public static User Create(string name, string email, UserRole role = UserRole.Guest)
        {
            var user = new User($"user_{++_counter}", name, email)
            {
                Role = role
            };
            return user;
        }

        public static bool ValidateEmail(string email)
        {
            return !string.IsNullOrEmpty(email) && email.Contains('@');
        }
    }

    /// <summary>
    /// Extension methods for User
    /// </summary>
    public static class UserExtensions
    {
        public static bool IsAdmin(this User user)
        {
            return user.Role == UserRole.Admin;
        }

        public static string ToJson(this User user)
        {
            return $"{{\"id\":\"{user.Id}\",\"name\":\"{user.Name}\"}}";
        }
    }
}

File: RepoPromptTests/CodeMap/Fixtures/java/smoke.java
// Smoke test fixture for Java codemap extraction
package com.example.user;

import java.util.*;
import java.util.concurrent.ConcurrentHashMap;
import java.util.stream.Collectors;

/**
 * User role enumeration
 */
public enum UserRole {
    ADMIN("admin"),
    EDITOR("editor"),
    VIEWER("viewer"),
    GUEST("guest");

    private final String value;

    UserRole(String value) {
        this.value = value;
    }

    public String getValue() {
        return value;
    }
}

/**
 * Data provider interface
 */
interface DataProvider<T> {
    List<T> fetch();
    void save(T item);
    boolean delete(String id);
}

/**
 * User data model
 */
class User {
    private final String id;
    private String name;
    private String email;
    private UserRole role;
    private final Date createdAt;

    public User(String id, String name, String email) {
        this.id = id;
        this.name = name;
        this.email = email;
        this.role = UserRole.GUEST;
        this.createdAt = new Date();
    }

    // Getters
    public String getId() { return id; }
    public String getName() { return name; }
    public String getEmail() { return email; }
    public UserRole getRole() { return role; }
    public Date getCreatedAt() { return createdAt; }

    // Setters
    public void setName(String name) { this.name = name; }
    public void setEmail(String email) { this.email = email; }
    public void setRole(UserRole role) { this.role = role; }

    public String getDisplayName() {
        return String.format("%s (%s)", name, role.getValue());
    }

    public boolean validate() {
        return id != null && !id.isEmpty() && name != null && !name.isEmpty();
    }

    @Override
    public String toString() {
        return "User{id='" + id + "', name='" + name + "'}";
    }
}

/**
 * Service configuration
 */
class Config {
    private int maxUsers;
    private int timeoutSeconds;

    public static final int DEFAULT_MAX_USERS = 1000;
    public static final int DEFAULT_TIMEOUT = 30;

    public Config() {
        this(DEFAULT_MAX_USERS, DEFAULT_TIMEOUT);
    }

    public Config(int maxUsers, int timeoutSeconds) {
        this.maxUsers = maxUsers;
        this.timeoutSeconds = timeoutSeconds;
    }

    public int getMaxUsers() { return maxUsers; }
    public int getTimeoutSeconds() { return timeoutSeconds; }
}

/**
 * User service for managing users
 */
class UserService implements DataProvider<User> {
    private final Map<String, User> users;
    private final Config config;
    
    private static UserService instance;

    public UserService(Config config) {
        this.users = new ConcurrentHashMap<>();
        this.config = config;
    }

    public static synchronized UserService getInstance(Config config) {
        if (instance == null) {
            instance = new UserService(config);
        }
        return instance;
    }

    @Override
    public List<User> fetch() {
        return new ArrayList<>(users.values());
    }

    @Override
    public void save(User user) {
        if (user == null) {
            throw new IllegalArgumentException("User cannot be null");
        }
        if (!user.validate()) {
            throw new IllegalArgumentException("Invalid user data");
        }
        if (users.size() >= config.getMaxUsers()) {
            throw new IllegalStateException("Max users reached");
        }
        users.put(user.getId(), user);
    }

    @Override
    public boolean delete(String id) {
        return users.remove(id) != null;
    }

    public Optional<User> find(String id) {
        return Optional.ofNullable(users.get(id));
    }

    public List<User> getByRole(UserRole role) {
        return users.values().stream()
            .filter(u -> u.getRole() == role)
            .collect(Collectors.toList());
    }

    public int count() {
        return users.size();
    }
}

/**
 * User factory class
 */
class UserFactory {
    private static int counter = 0;

    public static User create(String name, String email) {
        return create(name, email, UserRole.GUEST);
    }

    public static User create(String name, String email, UserRole role) {
        User user = new User("user_" + (++counter), name, email);
        user.setRole(role);
        return user;
    }

    public static boolean validateEmail(String email) {
        return email != null && email.contains("@");
    }
}
